### PR TITLE
Create password reset token form screen

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: packages/client/screens/PasswordResetConfirmationTokenScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                type: 'server',
+                message: 'Code is invalid.',
+            });
+        }, [passwordResetCodeIsInvalid, setError]);
+
+        function handlePasswordResetTokenSubmit({
+            code,
+        }: PasswordResetConfirmationTokenFormFieldValues) {
+            appService.send({
+                type: 'SUBMIT_PASSWORD_RESET_CONFIRMATION_FORM',
```
Comment (prastoin) [inline packages/client/screens/PasswordResetConfirmationTokenScreen.tsx:56]: thoughts: I'm not sure about this event name, unless I'm mistaken this event will perform a reset password token verification and not a password reset confirmation ( that, in my mind, could be the next step after validating a password reset code where user will have to define a new password )

File: packages/client/machines/appMachine.ts [hunk 2]
```diff
@@ -229,66 +232,175 @@ export function createAppMachine({
...
+
+                                            actions: appModel.assign({
+                                                email: (_, event) =>
+                                                    event.email,
+                                            }),
+                                        },
+                                    },
+                                },
 
                                 requestingPasswordReset: {
```
Comment (prastoin) [inline packages/client/machines/appMachine.ts:248]: Praise: great work ! nice to see a such well managed authentication machine

File: packages/client/__tests__/AuthenticationPasswordReset.test.tsx [hunk 3]
```diff
@@ -179,6 +182,154 @@ const passwordResetMachine =
...
+                                expect(
+                                    passwordResetCodeScreenContainer,
+                                ).toBeTruthy();
+                            });
+                        },
+                    },
+                    initial:
+                        'Displaying password reset successful request toast',
+                    states: {
+                        'Displaying password reset successful request toast': {
```
Comment (prastoin) [inline packages/client/__tests__/AuthenticationPasswordReset.test.tsx:202]: Question: why are we expecting a toast when entering the password reset workflow ?

File: packages/client/machines/appMachine.ts (hunk 2 above)
Comment (Devessier) [reply packages/client/machines/appMachine.ts:248]: Aha! Long live state machines!

File: packages/client/__tests__/AuthenticationPasswordReset.test.tsx (hunk 3 above)
Comment (Devessier) [reply packages/client/__tests__/AuthenticationPasswordReset.test.tsx:209]: I added a comment to explain the reason: https://github.com/AdonisEnProvence/MusicRoom/pull/553/commits/21dee2f90a6ebf7c1d3b40e4610ea61ce39d1cc7#diff-4982ec3e27a3d7f1d2d966a855028736a70f7956b076bf673165d46e9ebc89bcR202.

Please note the state is named _Password reset **code** form_. In this step we want to get the confirmation _code_.

**Review by prastoin — APPROVED** [review]
Great work ! ready to be merged
Thanks for the the tiny modifications

**Pull request merged**